packages/react-components/src/components/Drawer/DrawerMenuItem.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Icon, IIconProps, IStyle, mergeStyles, Stack } from '@fluentui/react';
+import React from 'react';
+import { useTheme } from '../../theming/FluentThemeProvider';
+import { BaseCustomStyles } from '../../types';
+import { submitWithKeyboard } from '../utils/keyboardNavigation';
+
+/**
+ * Props for the DrawerMenuItem
+ *
+ * @internal
+ */
+export interface _DrawerMenuItemProps {
+  onItemClick?: () => void;
```

Comment:
Should this callback be provided the `key` so that clients can use a single callback for all items instead of having to create individual closures.

They can still always ignore the `key` in the callback if they wish.

Reply:
Ah I wondered why fluent's contextual menu props had the itemKey in the callback (I was confused as you are passing in item key as a prop and onclick, it should be redundant to have itemKey returned in the callback) -- but what you say makes sense! I will add it in

Reply:
Added `ev` and `itemKey` as optional callback args to closely match IContextMenuProps (they take the whole item as the callback instead of key but I don't immediately see how that is done)